src/main/java/com/vroozi/service/SolrDataReindexService.java

```diff
@@ -0,0 +1,382 @@
+      }
+      solrReindexProcess.setProcessState(ProcessState.COMPLETED);
+    } else {
+      LOGGER.debug("Nothing to Process!");
+    }
+  }
+
+  /* Retries the failed batches */
+  public void processFailedBatches(List<Integer> unitIds) {
+    List<SolrReindexProcess> solrReindexProcesses = solrReindexProcessRepository
+        .findAllByProcessTypeAndProcessStateAndUnitIdIn(ProcessType.RETRY_FAILED,
+            ProcessState.UNPROCESSED, unitIds);
+
+    if (!CollectionUtils.isEmpty(solrReindexProcesses)) {
+      HttpSolrClient solr = new HttpSolrClient.Builder(solrUrl).build();
+      solr.setParser(new XMLResponseParser());
+
+      solrReindexProcesses.forEach(solrReindexProcess -> {
+        int unitId = solrReindexProcess.getUnitId();
+        if (!solrReindexProcess.getFailedRecords().isEmpty()) {
+
+          /* Fetching list of materialGroupMapping from db */
+          List<MaterialGroupMapping> materialGroupMappingList = materialGroupDao
+              .findByUnitIdOrderByCatalogCategoryCodeDesc(unitId);
+          solrReindexProcess.setProcessState(ProcessState.PROCESSING);
+          solrReindexProcess.setStartTime(new Date());
+          solrReindexProcessRepository.save(solrReindexProcess);
+          processingFailedBatchForUnitId(solr, solrReindexProcess, unitId,
+              materialGroupMappingList);
+        }
+
+        solrReindexProcess.setEndTime(new Date());
+        solrReindexProcess.setProcessState(ProcessState.COMPLETED);
+        solrReindexProcessRepository.save(solrReindexProcess);
+      });
+    }
+  }
+
+  private void processingFailedBatchForUnitId(HttpSolrClient solr,
+      SolrReindexProcess solrReindexProcess, int unitId,
+      List<MaterialGroupMapping> materialGroupMappingList) {
+    /* Processing failed batches against a unitId */
+    solrReindexProcess.getFailedRecords().forEach(failedRecord -> {
+      Result result = new Result(unitId);
+      try {
+        QueryResponse response = getQueryResponse(solr, failedRecord.getStart(),
+            failedRecord.getRecordsPerPage(), unitId);
+        List<SolrInputDocument> solrInputDocumentList = processSolrDocuments(unitId,
+            response, materialGroupMappingList, result);
+      /* Adding solrInputDocument to solr in batches, log the batches in db if solr.commit fails */
+        commitSolrBatch(solr, solrInputDocumentList);
+        solrReindexProcess.getResults().add(result);
+      } catch (Exception e) {
+        logFailedBatch(unitId, failedRecord.getStart(), failedRecord.getRecordsPerPage(),
+            ExceptionUtils.getStackTrace(e));
+        LOGGER.error("Exception occurred while retrying failed records ", e);
+      }
+    });
+  }
+
+  /* reindexes solr data for catalogItems against given unitId */
+  public void reindex(HttpSolrClient solr, SolrReindexProcess solrReindexProcess, Integer unitId) {
+    LOGGER.info("Going to start reindex data for unitId: {}", unitId);
```

Comment:
If we print a separator in logs before and after the processing for a unitId, it will make it more readable, like following
`LOGGER.debug("----------------------------------------------------------------------")`